crates/subspace-farmer/src/multi_farming.rs

```diff
@@ -35,16 +35,63 @@ impl MultiFarming {
         let mut commitments = Vec::with_capacity(plot_sizes.len());
         let mut farmings = Vec::with_capacity(plot_sizes.len());
 
-        for (plot_index, max_plot_pieces) in plot_sizes.into_iter().enumerate() {
-            let base_directory = base_directory.as_ref().join(format!("plot{plot_index}"));
-            std::fs::create_dir_all(&base_directory)?;
-            let (plot, subspace_codec, plot_commitments, farming) = farm_single_plot(
-                base_directory,
-                reward_address,
-                client.clone(),
-                max_plot_pieces,
-            )
-            .await?;
+        let mut results = plot_sizes
+            .into_iter()
+            .enumerate()
+            .map(|(plot_index, max_plot_pieces)| {
+                let base_directory = base_directory.as_ref().join(format!("plot{plot_index}"));
+                let client = client.clone();
+                async move {
+                    std::fs::create_dir_all(&base_directory)?;
+
+                    let identity = Identity::open_or_create(&base_directory)?;
+                    let public_key = identity.public_key().to_bytes().into();
+
+                    let plot_future = tokio::task::spawn_blocking({
+                        let base_directory = base_directory.clone();
+
+                        move || {
+                            // TODO: This doesn't account for the fact that node can
+                            // have a completely different history to what farmer expects
+                            info!("Opening plot");
+                            Plot::open_or_create(&base_directory, public_key, max_plot_pieces)
+                        }
+                    });
+
+                    let plot_commitments_future = tokio::task::spawn_blocking({
+                        let path = base_directory.join("commitments");
+
+                        move || {
+                            info!("Opening commitments");
+                            Commitments::new(path)
+                        }
+                    });
```

Comment:
Hm... the only reason this is async is because of these inherently synchronous calls wrapped into async tasks, while there are other sync calls like `std::fs::create_dir_all` and `Identity::open_or_create` above :confused: 

This should be just a function with parallel iterator with everything related to individual plot running synchronously and sequentially. I doubt it makes any noticeable difference if you run things related to one plot in parallel, only when we are talking about many plot in sequence it might be worthwhile to parallelize different plots.

Reply:
I just checked and it fails to spawn tasks inside the rayon threads (doesn't find the reactor).

Reply:
Should I remove the last commit so that pr would work? It seems that the rule of thumb here is not to spawn tasks within a rayon runtime.

Reply:
Indeed, hm... if rayon doesn't work, then IMHO it make the most sense to have regular iterator, wrap the whole thing into `tokio::task::spawn_blocking` and await in the second loop on individual entries without `FuturesUnordered` (with `Vec<JoinHandle>`).